### PR TITLE
Update firestore docs

### PR DESCRIPTION
**Pull request by TheIronDev**
Update v1 Firestore documentation

### CODE REVIEW

File: src/v1/providers/firestore.ts [hunk 1]
```diff
@@ -52,38 +49,32 @@ export function document<Path extends string>(path: Path) {
...
 }
 
-/** @hidden */
 // Multiple databases are not yet supported by Firestore.
 export function database(database: string) {
   return _databaseWithOptions(database, {});
 }
 
-/** @hidden */
 export function _databaseWithOptions(
```
Comment (TheIronDev) [inline src/v1/providers/firestore.ts:62]: IMO `_databaseWithOptions`, `_namespaceWithOptions`, `_documentWithOptions` can all be possibly removed as an export. `functions-builder` appears to use them, but we can clone-n-own in the separate file.

Thoughts

File: src/v1/providers/firestore.ts [hunk 2]
```diff
@@ -31,11 +31,8 @@ import * as logger from "../../logger";
 import { CloudFunction, Event, EventContext, makeCloudFunction } from "../cloud-functions";
 import { DeploymentOptions } from "../function-configuration";
 
-/** @hidden */
 export const provider = "google.firestore";
```
Comment (TheIronDev) [inline src/v1/providers/firestore.ts:34]: Maybe this is a personal nit but, shouldn't these three const strings be uppercased?

File: src/v1/providers/firestore.ts [hunk 3]
```diff
@@ -96,7 +87,6 @@ export class DatabaseBuilder {
 }
 
 export class NamespaceBuilder {
```
Comment (TheIronDev) [inline src/v1/providers/firestore.ts:89]: firestore.ts is the only thing use this. Does this need to be exported still?

Should it be @internal instread?

File: src/v1/providers/firestore.ts [hunk 4]
```diff
@@ -175,7 +163,6 @@ function changeConstructor(raw: Event) {
 }
 
 export class DocumentBuilder<Path extends string> {
```
Comment (TheIronDev) [inline src/v1/providers/firestore.ts:165]: Same thing with DocumentBuilder. This class is only used in this file. Remove `export` or `@internal`?

File: src/v1/providers/firestore.ts (hunk 1 above)
Comment (taeold) [reply src/v1/providers/firestore.ts:62]: Please mark this internal instead.

Let's try to keep the work to minimum - we don't necessarily have to fix things.

File: src/v1/providers/firestore.ts (hunk 2 above)
Comment (taeold) [reply src/v1/providers/firestore.ts:34]: Please mark this @internal instead.

if you mean variable name should be caps, maybe! if you mean the string value, i don't think so (they might map to eventflow event type)

File: src/v1/providers/firestore.ts (hunk 3 above)
Comment (taeold) [reply src/v1/providers/firestore.ts:89]: I think this object ultimately gets exposed on customer code.

Marking it @internal might break `tsc` on customer code 🤔

File: src/v1/providers/firestore.ts (hunk 4 above)
Comment (taeold) [reply src/v1/providers/firestore.ts:165]: Same thing - this is probably exposed to customer and must be exported and not be internal. (I could be wrong though).

**Review by taeold — CHANGES_REQUESTED** [review]

File: src/v1/providers/firestore.ts [hunk 5]
```diff
@@ -52,38 +49,32 @@ export function document<Path extends string>(path: Path) {
   return _documentWithOptions(path, {});
 }
 
-/** @hidden */
 // Multiple namespaces are not yet supported by Firestore.
 export function namespace(namespace: string) {
```
Comment (taeold) [inline src/v1/providers/firestore.ts:53]: If not hidden, can we try to documen this?

We can also consult w/ @egilmorez when he's back to document this section if you aren't sure how to document this.

File: src/v1/providers/firestore.ts (hunk 3 above)
Comment (TheIronDev) [reply src/v1/providers/firestore.ts:89]: 🤔  Leaving it then..

File: src/v1/providers/firestore.ts (hunk 5 above)
Comment (TheIronDev) [reply src/v1/providers/firestore.ts:56]: I don't have enough context on this..

File: src/v1/providers/firestore.ts (hunk 1 above)
Comment (TheIronDev) [reply src/v1/providers/firestore.ts:66]: Marked as internal it is!

File: src/v1/providers/firestore.ts (hunk 2 above)
Comment (TheIronDev) [reply src/v1/providers/firestore.ts]: renamed to caps and added `@internal`

File: src/v1/providers/firestore.ts [hunk 6]
```diff
@@ -31,9 +31,12 @@ import * as logger from "../../logger";
 import { CloudFunction, Event, EventContext, makeCloudFunction } from "../cloud-functions";
 import { DeploymentOptions } from "../function-configuration";
 
-export const provider = "google.firestore";
-export const service = "firestore.googleapis.com";
-export const defaultDatabase = "(default)";
+/** @internal */
+export const PROVIDER = "google.firestore";
```
Comment (taeold) [inline src/v1/providers/firestore.ts:35]: "technically" renaming an expose variable name is a breaking change.

Truthfully, I don't really care for that kind of breaking changes, but I'd err on the safe side and not change the casing here unless there are other practical purpose I'm missing.

File: src/v1/providers/firestore.ts (hunk 5 above)
Comment (taeold) [reply src/v1/providers/firestore.ts:56]: Let's leave it blank for now.

**Review by taeold — APPROVED** [review]
one minor nit on casing. otherwise lgtm.

File: src/v1/providers/firestore.ts (hunk 6 above)
Comment (TheIronDev) [reply src/v1/providers/firestore.ts]: Fair point, reverted this. 👍

File: src/v1/providers/firestore.ts (hunk 5 above)
Comment (TheIronDev) [reply src/v1/providers/firestore.ts:56]: kko!

File: src/v1/providers/firestore.ts (hunk 4 above)
Comment (TheIronDev) [reply src/v1/providers/firestore.ts:171]: SGTM!

**Pull request merged**